file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -129,84 +134,95 @@ protected internal override async Task OnExecuteCommand()
             IsCanceled = false;
             CancellationTokenSource = new CancellationTokenSource();
             CancellationToken cancellationToken = CancellationTokenSource.Token;
-            
+
             try
             {
-                logger.LogInformation("Execute job started for Action : {Action} and displayName :{displayName} .",Action,DisplayName);
+                logger.LogInformation("Execute job started for Action : {Action} and displayName :{displayName} .", Action, DisplayName);
                 var fileShareClient = fileShareClientFactory();
-                var buildBatchModel = BuildBatchModel();                
-                logger.LogInformation("File Share Service batch create started.");
-                batchHandle = await fileShareClient.CreateBatchAsync(buildBatchModel, CancellationToken.None); 
-                logger.LogInformation("File Share Service batch create completed for batch ID:{BatchId}.", batchHandle.BatchId);
-                FileUploadProgress.Clear();
-                try
+                var buildBatchModel = BuildBatchModel();
+                BatchSearchResponse batchSearchResponse = await SearchBatch();
+                logger.LogInformation($"{ batchSearchResponse.Count} duplicate batches found for action: {Action} and displayName: {DisplayName}.");
+                if (batchSearchResponse.Total > 0 && messageBoxService.ShowMessageBox($"Confirmation for displayName: {DisplayName}", $"{batchSearchResponse.Count} duplicate batches found. Do you still want to continue to execute the job ?",
+                MessageBoxButton.YesNo, MessageBoxImage.Question) == MessageBoxResult.No)
+                {
+                    logger.LogInformation($"File Share Service create new batch job cancelled for action: {Action} and displayName: {DisplayName}, because {batchSearchResponse.Count} duplicate batches found.");
+                    ExecutionResult = $"File Share Service create new batch cancelled to execute the job. ";
```

Comment:
Please change "File Share Service create new batch cancelled to execute the job. " to "File Share Service create new batch cancelled"

Reply:
updated as suggested. Thanks